tools/aiecc/aiecc/main.py

```diff
@@ -273,15 +273,36 @@ async def process_host_cgen(self):
           cmd += ['--gcc-toolchain=%s/usr' % opts.sysroot]
 
       thispath = os.path.dirname(os.path.realpath(__file__))
-      runtime_xaiengine_path = os.path.join(thispath, '..','..','runtime_lib', opts.host_target.split('-')[0], 'xaiengine')
-      xaiengine_include_path = os.path.join(runtime_xaiengine_path, "include")
-      xaiengine_lib_path = os.path.join(runtime_xaiengine_path, "lib")
+
+      # We can have differnet libxaie targets and need to point to the proper one
+      if (opts.host_target.split('-')[0] == 'x86_64'):
+        runtime_xaiengine_path = os.path.join(aiecc.configure.libxaie_x86_dir)
+        xaiengine_include_path = os.path.join(runtime_xaiengine_path, "include")
+        xaiengine_lib_path = os.path.join(runtime_xaiengine_path, "lib")
+      else:
```

Comment:
This seems odd.   Shouldn't we just compile the right libXAIE when targetting VK5000?  (i.e. handle this difference at configuration time and put the right libxaie in runtime_lib/x86_64

Reply:
I agree the removed lines are broken. I often need to patch these lines manually. The reason the removed code is broken is because it hard codes a libxaie location whereas cmake lets the user pick custom locations in order to support different platforms. It is also broken when the libxaie path in the build environment does not match the path in the deployed environment.

I think the general fix is to provide a runtime override like this patch does, but without tying it to a specific host target. The fallback should be the cmake configured path, which itself defaults to the currently hard coded path.

Reply:
I have the version of aie-rt here which combines the CDO changes and our PCIe backend in my fork: https://github.com/eddierichter-amd/aie-rt/tree/converged-aie-rt. This should then be used as the submodule for aie-rt and it can be compiled with the proper flags.